pkg/operator/resource/resourceapply/monitoring.go

```diff
@@ -96,6 +96,65 @@ func ApplyServiceMonitor(ctx context.Context, client dynamic.Interface, recorder
 	return newObj, true, err
 }
 
+var prometheusGVR = schema.GroupVersionResource{Group: "monitoring.coreos.com", Version: "v1", Resource: "prometheuses"}
+
+// ApplyPrometheus applies the Prometheus object.
+func ApplyPrometheus(ctx context.Context, client dynamic.Interface, recorder events.Recorder, required *unstructured.Unstructured) (*unstructured.Unstructured, bool, error) {
+	namespace := required.GetNamespace()
+
+	existing, err := client.Resource(prometheusGVR).Namespace(namespace).Get(ctx, required.GetName(), metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		newObj, createErr := client.Resource(prometheusGVR).Namespace(namespace).Create(ctx, required, metav1.CreateOptions{})
+		if createErr != nil {
+			recorder.Warningf("PrometheusCreateFailed", "Failed to create Prometheus.monitoring.coreos.com/v1: %v", createErr)
+			return nil, true, createErr
+		}
+		recorder.Eventf("PrometheusCreated", "Created Prometheus.monitoring.coreos.com/v1 because it was missing")
+		return newObj, true, nil
+	}
+	if err != nil {
+		return nil, false, err
+	}
+
+	existingCopy := existing.DeepCopy()
+
+	toUpdate, modified, err := ensureGenericSpec(required, existingCopy, noDefaulting, equality.Semantic)
```

Comment:
I'm not actually sure `ensureGenericSpec` is smart enough, or if I need to bring over some of [this defaulting logic][1].  I also don't see any `ObjectMeta` comparisons going on in the functions from this file that I pattern-matched on, so I'm not clear if we will notice and stomp unacceptable annotation/label changes.

[1]: https://github.com/openshift/cluster-monitoring-operator/commit/ef5f348a939419818f9da862ff554af8853c9738#diff-2895f30116602d8fe6b0545c186d4f81247fa4eaab049dab174c6c91c0036e08R639-R667